examples/nodejs/lambda-examples/cloudwatch-metrics/infrastructure/lib/momento-metrics-stack.ts

```diff
@@ -0,0 +1,337 @@
+
+export class MomentoMetricsStack extends cdk.Stack {
+  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
+    super(scope, id, props);
+
+    // Step 1: read in environment variables from your .env file
+    if (!this.validateMomentoApiKey(process.env.MOMENTO_API_KEY)) {
+      throw new Error('Missing required environment variable MOMENTO_API_KEY');
+    }
+    const momentoApiKey = String(process.env.MOMENTO_API_KEY);
+
+    if (!this.validateStackConfig(process.env.EXAMPLE_MOMENTO_APPLICATION)) {
+      throw new Error('Missing or invalid entry for required environment variable EXAMPLE_MOMENTO_APPLICATION');
+    }
+    const stackConfig = String(process.env.EXAMPLE_MOMENTO_APPLICATION);
+
+    if (!this.validateLogGroupName(stackConfig, process.env.LOG_GROUP_NAME)) {
+      throw new Error('Missing required environment variable LOG_GROUP_NAME');
+    }
+    const dashboardOnlyLogGroupName = String(process.env.LOG_GROUP_NAME);
+
+    // Step 2: insert your Momento API key into AWS Secrets Manager
+    const apiKeySecret = new secrets.Secret(this, 'MomentoMetricsApiKey', {
+      secretName: 'MomentoMetricsApiKey',
+      secretStringValue: new cdk.SecretValue(momentoApiKey),
+    });
+
+    /*
+    Step 3: determine which log group name to use when creating our log group.
+    Note: Lambda functions automatically create a log group titled 'aws/lambda/FunctionName',
+    so if you create only the dashboard here and use your own Lambda function, make sure to
+    provide your function's log group name for the ___ parameter.
+    */
+    const configToLogGroupName = new Map([
+      ["lambda", '/aws/lambda/MomentoMetricsMiddlewareCDKExample'],
+      ["ecs", '/aws/ecs/MomentoMetricsMiddlewareCDKExample'],
+      ["dashboard-only", dashboardOnlyLogGroupName]
+    ]);
+    const logGroupName = configToLogGroupName.get(stackConfig);
+
+    const logGroup = new LogGroup(this, 'Logs', {
+      logGroupName: logGroupName,
+      retention: RetentionDays.ONE_DAY,
+    });
+
+    // Step 4: create the Lambda or ECS example as specified. Otherwise skip this step
+    // and create only the dashboard.
+    switch (stackConfig) {
+      case "lambda": {
+        this.setUpLambdaFunction(apiKeySecret);
+        break;
+      }
+      case "ecs": {
+        this.setUpEcsCluster(logGroup, apiKeySecret);
+        break;
+      }
+      case "dashboard-only": {
+        console.log('Skipping to dashboard creation');
+        break;
+      }
+      default: {
+        throw new Error('Unimplemented CDK stack application');
+      }
+    }
+
+    /* Step 5:
+    The Momento experimental metrics middleware produces JSON logs with metrics about each Momento request. An example log entry looks like this:
+        (Momento: _ExperimentalMetricsLoggingMiddleware): 
+        {
+            "numActiveRequestsAtStart": 1,
+            "numActiveRequestsAtFinish": 1,
+            "requestType": "MiddlewareMessage",
```

Comment:
woof, is this what really shows up for the request type now? that seems like a bug in my last refactor. We want this field to show something like `Get`, `Set`, `ListPushBack`, etc. If all of the requests are showing up with just this, then we should make a quick tweak on `main` to fix that and then rebase your branch. I can help with that if you need another pair of eyes.

Reply:
Good catch! Got a PR for this: #993